From 54d4a0d341cdcf38b0bc716367f707d142f21c44 Mon Sep 17 00:00:00 2001
From: Rander Wang <rander.wang@linux.intel.com>
Date: Mon, 23 Jul 2018 15:38:54 +0800
Subject: [PATCH 1357/1582] ASoC: SOF: fix range check for max channels

Fix the check for maximum number of channels to be in the correct range.

Signed-off-by: Rander Wang <rander.wang@linux.intel.com>
---
 sound/soc/sof/topology.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index e3ac2f2..288eba6 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -248,7 +248,7 @@ static int sof_control_load_volume(struct snd_soc_component *scomp,
 	struct sof_ipc_ctrl_data *cdata;
 
 	/* validate topology data */
-	if (le32_to_cpu(mc->num_channels) >= SND_SOC_TPLG_MAX_CHAN)
+	if (le32_to_cpu(mc->num_channels) > SND_SOC_TPLG_MAX_CHAN)
 		return -EINVAL;
 
 	/* init the volume get/put data */
-- 
2.7.4

